File: spigot/src/main/java/io/github/toberocat/improvedfactions/listener/PlayerJoinListener.java
package io.github.toberocat.improvedfactions.listener;

import io.github.toberocat.improvedFactions.core.player.FactionPlayer;
import io.github.toberocat.improvedFactions.core.registry.ImplementationHolder;
import io.github.toberocat.improvedfactions.MainIF;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.player.PlayerJoinEvent;
import org.jetbrains.annotations.NotNull;

public class PlayerJoinListener extends SpigotEventListener {

    public PlayerJoinListener(@NotNull MainIF plugin) {
        super(plugin);

        Bukkit.getOnlinePlayers().forEach(this::join);
    }

    @EventHandler
    private void event(PlayerJoinEvent event) {
        join(event.getPlayer());
    }

    private void join(@NotNull Player player) {
        FactionPlayer<?> factionPlayer = plugin.getPlayer(player.getUniqueId());
        if (factionPlayer == null) return;

        ImplementationHolder.playerJoin(factionPlayer);
    }
}

File: improved-factions/src/main/kotlin/io/github/toberocat/improvedfactions/listeners/PlayerJoinListener.kt
package io.github.toberocat.improvedfactions.listeners

import io.github.toberocat.improvedfactions.database.storage.StorageManager
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent

class PlayerJoinListener : Listener {

    @EventHandler
    private fun onJoin(event: PlayerJoinEvent) {
        val player: Player = event.player

        StorageManager.persistKnownPlayer(player.uniqueId, player.name)
    }
}
